lighthouse-core/config/budget.js

```diff
@@ -240,6 +240,38 @@ class Budget {
     };
   }
 
+  /**
+   * @param {string} origin
+   * @return {string}
+   */
+  static validateOrigin(origin) {
+    if (origin.endsWith(`/`)) {
+      throw new Error(`Invalid origin: ${origin}. Origin should not contain a trailing '/'.`);
+    }
+    let verifiedOrigin;
+    try {
+      verifiedOrigin = new URL(origin).origin;
+    } catch (err) {
+      throw new Error(`${origin} is not a valid URL.`);
+    }
+    if (origin !== verifiedOrigin) {
+      throw new Error(`Invalid origin: ${origin}. Did you mean ${verifiedOrigin}?`);
+    }
+    return origin;
+  }
+
+  /**
+   * @param {unknown} origins
+   * @return {undefined|Array<string>}
+   */
+  static validateFirstPartyOrigins(origins) {
+    if (Array.isArray(origins) && origins.every(origin => typeof origin === 'string')) {
+      return origins.map(this.validateOrigin);
```

Comment:
```suggestion
      return origins.map(Budget.validateOrigin);
```

we don't do any overriding of this class or anything that necessitates `this` do we?